src/twisted/web/test/test_http.py

```diff
@@ -3325,6 +3325,41 @@ def test_expect100ContinueHeader(self):
 
 
 
+class ConnectRequestTests(unittest.TestCase, ResponseTestMixin):
+    """
+    Test correct RFC 7231 handling for CONNECT requests. Responses to these
+    tests must not contain Transfer-Encoding or Content-Length headers
+    """
+
+    def test_headers(self):
+        """
+        A HTTP connection is opened and a CONNECT request is received, which
+        should return a 200 response with the HTTP version header, the host and
+        port as well as the CONNECT command but not a Transfer-Encoding or
+        Content-Length header, which both are not allowed in a 200 response.
+        """
+        transport = StringTransport()
+        channel = http.HTTPChannel()
+        channel.requestFactory = DummyHTTPHandlerProxy
+        channel.makeConnection(transport)
+        channel.dataReceived(b"CONNECT example.org:443 HTTP/1.1\r\n")
+        channel.dataReceived(b"Host: example.org:443\r\n")
+
+        self.assertEqual(transport.value(), b"")
+        channel.dataReceived(b"\r\n")
+
+        self.assertTrue(
+            transport.value().startswith(b"HTTP/1.1 200 OK\r\n"))
```

Comment:
This assertion looks redundant with the following.  Is there a particular reason to have it?  If not, I think it should be removed.  The `assertResponseEquals` will cover this check.